util/junitxml/junitxml_test.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2024 Kushal Shukla
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package junitxml
+
+import (
+	"bytes"
+	"errors"
+	"testing"
+
+	"github.com/grafana/regexp"
+)
+
+func TestJunitOutput(t *testing.T) {
```

Comment:
nit: this is pretty weird to test XML validity with regexp. I would have expected either a direct compare to a reference string or using encoding/xml to parse and validate.

Reply:
what im thinking like , can't we convert the yaml file data into struct data and i already have struct data as per encoding/xml package and then we can compare them .